### PR TITLE
Implement processes operations

### PR DESCRIPTION
**Pull request by khaledkbadr**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ type Option interface {
 }
 
 type JobId string
+type ProcessId string
```
Comment (muhamadazmy) [inline client.go]: ProcessID `PID` is always an int

File: core.go [hunk 2]
```diff
@@ -24,15 +24,28 @@ type JobStats struct {
 	Debug string  `json:"debug,ommitempty"`
 }
 
+type Process struct {
+	Command    string `json:"cmdline"`
+	Createtime uint64 `json:"createtime"`
+	PID        uint64 `json:"pid"`
```
Comment (muhamadazmy) [inline core.go]: Use `ProcessID` type u defined above

File: core.go [hunk 3]
```diff
@@ -24,15 +24,28 @@ type JobStats struct {
...
+	PPID       uint64 `json:"ppid"`
```
Comment (muhamadazmy) [inline core.go]: Same as PID

File: core.go [hunk 4]
```diff
@@ -24,15 +24,28 @@ type JobStats struct {
...
+	VMS        uint64 `json:"vms"`
+}
+
 type CoreManager interface {
 	System(cmd string, env map[string]string, cwd string, stdin string, opt ...Option) (JobId, error)
 	SystemArgs(cmd string, args []string, env map[string]string, cwd string, stdin string, opt ...Option) (JobId, error)
 	Bash(bash string, stdin string, opt ...Option) (JobId, error)
 	Ping() error
 	Jobs() ([]Job, error)
 	Job(job JobId) (*Job, error)
```
Comment (muhamadazmy) [inline core.go]: Where is the `Job` struct defined, or renamed?

Comment (khaledkbadr) [inline core.go]: https://github.com/g8os/go-client/pull/7/files#diff-bb6c2025258b800c449511ecd396c87aL10

**Pull request merged**